# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/indexedDB/ActorsChild.h
# Commit: d2772a7482ed
# Full Hash: d2772a7482ed9394a2c30acc6d03caa36317e75f
# Author: Simon Giesecke <sgiesecke@mozilla.com>
# Date: 2019-09-30 16:21:37
# Regressor Bug: 1168606
# File Overlap Count: 1
# Description:
#   Bug 1168606 - Allow current key to be provided with PBackgroundIDBCursor::Continue. r=ttung,asuth
#   
#   The key is assumed to be unset for now.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D43251
# ==============================================================================

diff -r 340dfa69aed8 -r d2772a7482ed dom/indexedDB/ActorsChild.h
--- a/dom/indexedDB/ActorsChild.h	Fri Sep 27 10:11:45 2019 +0000
+++ b/dom/indexedDB/ActorsChild.h	Fri Sep 27 10:11:41 2019 +0000
@@ -649,7 +649,8 @@
     NS_ASSERT_OWNINGTHREAD(BackgroundCursorChild);
   }
 
-  void SendContinueInternal(const CursorRequestParams& aParams);
+  void SendContinueInternal(const CursorRequestParams& aParams,
+                            const Key& aCurrentKey);
 
   void SendDeleteMeInternal();
 
@@ -700,8 +701,10 @@
   virtual mozilla::ipc::IPCResult RecvResponse(
       const CursorResponse& aResponse) override;
 
+ public:
   // Force callers to use SendContinueInternal.
-  bool SendContinue(const CursorRequestParams& aParams) = delete;
+  bool SendContinue(const CursorRequestParams& aParams,
+                    const Key& aCurrentKey) = delete;
 
   bool SendDeleteMe() = delete;
 };